### ISSUE TITLE
DL4J Keras Import: 2.2.x/2.3.x compatibility

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/deeplearning4j#6550]
Reported by @Kissy: https://github.com/deeplearning4j/deeplearning4j/issues/6527#issuecomment-427737778

> I reproduce the exception compiling with Keras 2.2.3 and importing with 1.0.0-beta2.
It seems that the keras field for "config" (retrieve here modelConfig.get(config.getModelFieldConfig())) has changed from list of object in version 2.0.X to object representing map.

> Keras 2.0.9 :
> `{
  "backend": "tensorflow",
  "keras_version": "2.0.9",
  "class_name": "Sequential",
  "config": [
    {
      "class_name": "Dense",
      "config": {
        "kernel_initializer": {
          "class_name": "VarianceScaling",
          "config": {
            "distribution": "uniform",
            "scale": 1.0,
            "seed": null,
            "mode": "fan_avg"
          }`

> Keras 2.2.3 :
> `{
  "backend": "tensorflow",
  "keras_version": "2.2.3",
  "class_name": "Sequential",
  "config": {
    "name": "sequential_1",
    "layers": [
      {
        "class_name": "Dense",
        "config": {
          "name": "dense_1",
          "trainable": true,
          "batch_input_shape": [
            null,
            100
          ],`

**Comment by maxpumperla** [comment]
that's disgusting... alright, let me patch this. I hope this doesn't happen again.

**Comment by timmolter** [comment]
I just did a test working backwards from keras 2.2.4 and there is no issue for 2.2.2. 

Just an INFO log about `Unexpected end-of-input`:

```
[main] INFO org.deeplearning4j.nn.modelimport.keras.Hdf5Archive - Unexpected end-of-input: was expecting closing '"' for name
```

I could post the two different version h5 files for cifar if you want (see issue #6577).

**Comment by maxpumperla** [comment]
@timmolter that's fine, we're on top of this issue. thanks for offering help

**Comment by maxpumperla** [comment]
fix will be available on snapshots starting tomorrow

**Comment by timmolter** [comment]
use `1.0.0-SNAPSHOT`.

**Comment by AlexDBlack** [comment]
@bugfoot not from maven, but from sonatype.
See this page: https://deeplearning4j.org/docs/latest/deeplearning4j-config-snapshots
In the future, ask these sorts of questions on the gitter channel, not closed issues - you'll get a quicker response there.
https://gitter.im/deeplearning4j/deeplearning4j
